app/src/main/java/de/danoeh/antennapod/fragment/QueueFragment.java

```diff
@@ -324,38 +339,52 @@ public void onConfirmButtonPressed(
                     ((MainActivity) requireActivity()) .loadChildFragment(
                             EpisodesApplyActionFragment.newInstance(queue, ACTION_DELETE | ACTION_REMOVE_FROM_QUEUE));
                     return true;
+                case R.id.queue_sort_automatically:
+                    boolean sortedAutomaticallyOld = UserPreferences.isQueueSortedAutomatically();
+                    boolean sortedAutomaticallyNew = !sortedAutomaticallyOld;
+                    if (sortedAutomaticallyNew) {
+                        // We have to choose an initially sort order, let's sort by episode date
+                        UserPreferences.QueueSortOrder sortOrder = UserPreferences.QueueSortOrder.DATE_NEW_OLD;
+                        UserPreferences.setQueueSortOrder(sortOrder);
+                        QueueSorter.sort(QueueSorter.queueSortOrder2Rule(sortOrder), true);
+                    } else {
+                        UserPreferences.setQueueSortOrder(UserPreferences.QueueSortOrder.MANUALLY);
+                    }
+                    // Update sort menu items visibility and state
+                    getActivity().invalidateOptionsMenu();
```

Comment:
Here, the list needs to be invalidated, too. Otherwise, the drag handles are only hidden when pausing+resuming the app.

Reply:
Good hint, that's still a bug in my code.

Reply:
Bug fixed by your suggestion below.